### PR TITLE
T2D-168 Default to return only last version when querying by accession

### PR DESCRIPTION
**Pull request by jorizci**
Depends on #29

### CODE REVIEW

File: src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicAccessioningService.java [hunk 1]
```diff
@@ -147,9 +148,28 @@ public BasicAccessioningService(AccessionGenerator<MODEL, ACCESSION> accessionGe
     }
 
     @Override
-    public List<AccessionWrapper<MODEL, HASH, ACCESSION>> getByAccessions(List<ACCESSION> accessions,
-                                                                          boolean hideDeprecated) {
-        return dbService.findAllAccessionMappingsByAccessions(accessions);
+    public List<AccessionWrapper<MODEL, HASH, ACCESSION>> getByAccessionIds(List<ACCESSION> accessions,
```
Comment (jmmut) [inline src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicAccessioningService.java]: I'm not sure I understand the value of renaming "accessions" to "accessionIds", especially now that the same accession is repeated in several versions

Comment (jorizci) [reply src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicAccessioningService.java]: I'm not completely sure either, currently there are mentions to accessions as in the full object or accession mappings or wrappers. We should probably decide a terminology and enforce it.

File: src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicAccessioningService.java [hunk 2]
```diff
@@ -147,9 +148,28 @@ public BasicAccessioningService(AccessionGenerator<MODEL, ACCESSION> accessionGe
...
+                                                                            boolean hideDeprecated) {
+        final List<AccessionWrapper<MODEL, HASH, ACCESSION>> allAccessionMappingsByAccessions
+                = dbService.findAllAccessionMappingsByAccessions(accessions);
+        Map<ACCESSION, Integer> accessionToMaxVersion = generateAccessionToMaxVersion(allAccessionMappingsByAccessions);
```
Comment (selvaebi) [inline src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicAccessioningService.java]: can the function name be getMapOfAccessionToMaxVersion instead generateAccessionToMaxVersion ?

Comment (jorizci) [reply src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicAccessioningService.java]: It is not a getter, processes the current information and generates the map.

Comment (selvaebi) [reply src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicAccessioningService.java]: ahh sorry in that case generateMapOfAccessionToMaxVersion ?

Comment (jorizci) [reply src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicAccessioningService.java]: No, you already know it is a map from the return. Adding that information to the method name is a little bit redundant.

**Pull request merged**